src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1916,28 +1846,26 @@ private static void VerifyPseudoCustomAttribute(RuntimeType pca)
         #endregion
 
         #region Internal Static
-        internal static Attribute[] GetCustomAttributes(RuntimeType type, RuntimeType caType, out int count)
+        internal static RuntimeType.ListBuilder<Attribute> GetCustomAttributes(RuntimeType type, RuntimeType caType)
         {
             Debug.Assert(type != null);
             Debug.Assert(caType != null);
 
-            count = 0;
-
             bool all = caType == typeof(object) || caType == typeof(Attribute);
             if (!all && !s_pca.ContainsKey(caType))
-                return null;
+                return default;
 
-            Attribute[] pcas = new Attribute[all ? 2 : 1];
+            RuntimeType.ListBuilder<Attribute> pcas = new RuntimeType.ListBuilder<Attribute>(all ? 2 : 1);
```

Comment:
It is pretty unlikely that the type will have pseudo custom attributes. This should rather start with the default (zero size) and leave the rare case where the type has more than one pseudo custom attribute do a bit more extra copying.

Reply:
(Same for other places)

Reply:
Agreed, changing it up - slightly amusing aside: one instance of `all ? 4 : 1` was checking for a parameter being both `in` and `out` below as 2 of the 4.

Reply:
`[In,Out]` is actually perfectly legal and not that unusual for interop.

Reply:
Hmm true, TIL - can it also be optional in that case though?

Reply:
Yes, `[In]`/`[Out]` are optional. Interop has defaults for these that work fine for most cases. An explicit `[In]` and/or `[Out]` is only required to override the default.